ion-schema/src/model/constraints/annotations_v2_simple.rs

```diff
@@ -0,0 +1,260 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::internal_traits::*;
+use crate::ion_extension::SymbolExtensions;
+use crate::model::builder::TypeDefinitionBuilder;
+use crate::model::constraints::AnnotationsV2Modifier::{Closed, ClosedAndRequired, Required};
+use crate::result::{invalid_schema_error, invalid_schema_error_raw, IonSchemaResult};
+use crate::{IonSchemaElement, ViolationInfo, ViolationRecorder, ISL_2_0};
+use ion_rs::{Element, Symbol, ValueWriter};
+use std::ops::ControlFlow;
+use std::sync::Arc;
+
+/// Modifier for the simple syntax of Ion Schema 2.0 `annotations` constraint.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum AnnotationsV2Modifier {
+    Closed,
+    Required,
+    ClosedAndRequired,
+}
+impl AnnotationsV2Modifier {
+    fn is_closed(&self) -> bool {
+        *self != AnnotationsV2Modifier::Required
+    }
+    fn is_required(&self) -> bool {
+        *self != AnnotationsV2Modifier::Closed
+    }
+}
+
+/// Represents the [annotations] constraint, simple syntax, in Ion Schema 2.0
+///
+/// [annotations]: https://amazon-ion.github.io/ion-schema/docs/isl-2-0/spec#annotations
+#[derive(Debug, PartialEq, Clone)]
+pub struct AnnotationsV2Simple {
+    pub(crate) modifier: AnnotationsV2Modifier,
+    pub(crate) annotations: Vec<Symbol>,
+}
+
+impl AnnotationsV2Simple {
```

Comment:
Are we only thinking about having immutable constraints here right now? I was wondering if we should add something like `add_annotation` or `with_annotations` to push a new annotation to this constraint. And maybe even remove. All the APIs to modify the annotation constraint.

Reply:
I've created https://github.com/amazon-ion/ion-schema-rust/issues/233